packages/react/react/src/hooks/useStorageState.ts

```diff
@@ -53,12 +52,12 @@ export function useStorageState<T>(
     }
   }, [defaultValue, key, storage]);
 
-  const [state, setState] = useState<Serializable<T> | undefined>(getValue);
+  const [state, setState] = useState<T | undefined>(getValue);
 
   const set = useCallback(
-    (value: SetStateAction<Serializable<T> | undefined>) => {
+    (value: SetStateAction<T | undefined>) => {
       setState(curr => {
-        const nextValue = typeof value === 'function' ? value(curr) : value;
+        const nextValue = value instanceof Function ? value(curr) : value;
```

Comment:
Is there a reason a `typeof` check is replaced with `instanceof` here?

Reply:
This change was caused by the mistake of generic constraint only on the first function among the overloading functions.

 If generic T is not constrained to extend serializable, T will not be callable, so I modified it like this. 

Since I fixed the generic constraint from the above conversation to be added to all overloaded functions, this change can be reversed.

Reply:
Thanks! What about reverting this change and making this pull request more focused on the main idea?

Reply:
I reverted the change as you said. Thank you for taking the time to review my pull request and provide feedback!